File: deployment/gunfight_launcher/requirements/game_launcher/source/gunFightLauncher/remote_game_instance/rect.py
class Rect:
    def __init__(self, x, y, width, height):
        self.x = x
        self.y = y
        self.width = width
        self.height = height

    def __str__(self):
        return f'({self.x}, {self.y}, {self.width}, {self.height})'

    def collide(self, other):
        return (self.x + self.width >= other.x and
                self.x <= other.x + other.width and
                self.y + self.height >= other.y and
                self.y <= other.y + other.height)

    def get_center(self):
        return {"x": self.x + self.width / 2, "y": self.y + self.height / 2}

File: deployment/front-end/source/front-end/all-games-fronts/game-srcs/classes/Rect.js
export default class Rect {
    constructor(posX, posY, width, height) {
        this.posX = posX;
        this.posY = posY;
        this.width = width;
        this.height = height;
    }

    static center(screenWidth, screenHeight, width, height) {
        const screenCenter = {x: screenWidth / 2, y: screenHeight / 2};

        return new Rect(screenCenter.x - width / 2, screenCenter.y - height / 2, width, height);
    }

    collide(other) {
        return (this.posX + this.width >= other.posX &&
            this.posX <= other.posX + other.width &&
            this.posY + this.height >= other.posY &&
            this.posY <= other.posY + other.height)
    }

    getCenterCoords() {
        return {
            x: this.posX + this.width / 2,
            y: this.posY + this.height / 2
        }
    }
}